plugins/woocommerce/tests/e2e-pw/tests/shopper/my-account-downloads.spec.js

```diff
@@ -0,0 +1,123 @@
+	let productId, orderId;
+
+	test.beforeAll( async ( { baseURL } ) => {
+		const api = new wcApi( {
+			url: baseURL,
+			consumerKey: process.env.CONSUMER_KEY,
+			consumerSecret: process.env.CONSUMER_SECRET,
+			version: 'wc/v3',
+		} );
+		// add product
+		await api
+			.post( 'products', {
+				name: product.name,
+				type: 'simple',
+				regular_price: '9.99',
+				downloadable: true,
+				downloads: [
+					{
+						name: 'Test file',
+						file: `${ baseURL }/test-file/`,
+					},
+				],
+				download_limit: 1,
+			} )
+			.then( ( response ) => {
+				productId = response.data.id;
+			} );
+		// create customer
+		await api
+			.post( 'customers', customer )
+			.then( ( response ) => ( customer.id = response.data.id ) );
+		// create an order
+		await api
+			.post( 'orders', {
+				set_paid: true,
+				status: 'completed',
+				billing: {
+					first_name: 'Jane',
+					last_name: 'Smith',
+					email: customer.email,
+				},
+				line_items: [
+					{
+						product_id: productId,
+						quantity: 1,
+					},
+				],
+			} )
+			.then( ( response ) => {
+				orderId = response.data.id;
+			} );
+		// once the order is created, assign it to our existing customer user
+		await api.put( `orders/${ orderId }`, {
+			customer_id: customer.id,
+		} );
+	} );
+
+	test.afterAll( async ( { baseURL } ) => {
+		const api = new wcApi( {
+			url: baseURL,
+			consumerKey: process.env.CONSUMER_KEY,
+			consumerSecret: process.env.CONSUMER_SECRET,
+			version: 'wc/v3',
+		} );
+		await api.delete( `products/${ productId }`, {
+			force: true,
+		} );
+		await api.delete( `orders/${ orderId }`, { force: true } );
+		await api.delete( `customers/${ customer.id }`, { force: true } );
+	} );
+
+	test( 'can see downloadable file and click to download it', async ( {
+		page,
+	} ) => {
+		await page.goto( 'my-account/downloads/' );
+		// sign in as the "customer" user
+		await page.locator( '#username' ).fill( customer.username );
+		await page.locator( '#password' ).fill( customer.password );
+		await page.locator( 'text=Log in' ).click();
+
+		// verify that the downloadable product exist in downloads
+		await expect( page.locator( 'h1.entry-title' ) ).toContainText(
+			'Downloads'
+		);
+		await expect( page.locator( 'td.download-product' ) ).toContainText(
+			product.name
+		);
+		await expect( page.locator( 'td.download-remaining' ) ).toContainText(
+			product.downloadLimit
+		);
+		await expect(
+			page.locator( '.woocommerce-MyAccount-downloads-file' )
+		).toContainText( 'Test file' );
+
+		// click to simulate downloading and verify the file doesn't exist anymore in downloads
+		await page.locator( '.woocommerce-MyAccount-downloads-file' ).click();
+		await page.goto( 'my-account/downloads/' );
+		await expect( page.locator( '.woocommerce-info' ) ).toContainText(
+			'No downloads available yet.'
+		);
+		await expect( page.locator( '.wc-forward' ) ).toContainText(
+			'Browse products'
+		);
```

Comment:
OK, after seeing these assertions, I now see why a different customer will make sense, and why you went with this approach. If we'll use the pre-existing customer `Jane Smith`, these assertions might fail because other tests might add their own orders with downloadable products to this customer, so the 'No downloads available yet.' and 'Browse products' texts won't appear. So yeah, this is a good argument to using a different customer.

However, I don't think there are other e2e tests that add orders with downloadable products to the `Jane Smith` customer so I guess it should still be safe to use it? Alternatively we can use safer assertions like verifying that the "Test file" downloadable product is no longer visible after simulating the download.

Reply:
Hi @rodelgc 
Thanks so much for a great review!

I always try to keep tests independent and in isolation as much as possible, so we don't end up situation like "hey, why I don't see this data anymore, I expected it to be present". If we keep creating data to test within the test, we are completely independent. Longer execution is solvable by parallelism. Please let me know if it makes sense to you in which case we can ignore above comments. This can be good topic to discuss on a team level as well.

Reply:
Yeah, I believe it's fine to use a different test customer in this case because aside from making this truly independent, I noticed from our list of new critical flows (or critical functionalities) that there could be new flows related to product downloads and that they might conflict with this test if the pre-existing customer won't be used here. 

I'll mark the conversations above as resolved. Thanks for adding this much needed test BTW @veljkho !